code/game/atoms.dm

```diff
@@ -2079,11 +2079,19 @@
 
 	return TRUE
 
-//Update the screentip to reflect what we're hoverin over
 /atom/MouseEntered(location, control, params)
-	. = ..()
+	SSmouse_entered.hovers[usr.client] = src
+
+/// Fired whenever this atom is the most recent to be hovered over in the tick.
+/// Preferred over MouseEntered if you do not need information such as the position of the mouse.
+/// Especially because this is deferred over a tick, do not trust that `client` is not null.
+/atom/proc/on_mouse_enter(client/client)
+	SHOULD_NOT_SLEEP(TRUE)
+
+	var/mob/user = client?.mob
```

Comment:
How expensive is the ? here and below?

Reply:
JOHN ITS TOO LATE FOR THIS

Reply:
It's non-negotiable code (client can always be null, but *especially* after a tick has passed, which is when this fires) so it doesn't matter